pkg/reconciler/managed/reconciler.go

```diff
@@ -89,6 +96,217 @@ func ControllerName(kind string) string {
 	return "managed/" + strings.ToLower(kind)
 }
 
+// ManagementPoliciesChecker is used to perform checks on management policies
+// to determine specific actions are allowed, or if they are the only allowed
+// action.
+type ManagementPoliciesChecker interface {
+	// IsSupported returns true if the management policy is supported.
+	IsSupported() bool
+	// IsDisabledNonDefault returns true if the management policy is set to a non-default
+	// value, and the management policy feature is disabled. Used as a sanity
+	// check.
+	IsDisabledNonDefault() bool
+	// IsPaused returns true if the resource is paused based
+	// on the management policy.
+	IsPaused() bool
+
+	// ShouldOnlyObserve returns true if only the Observe action is allowed.
+	ShouldOnlyObserve() bool
+	// ShouldCreate returns true if the Create action is allowed.
+	ShouldCreate() bool
+	// ShouldLateInitialize returns true if the LateInitialize action is
+	// allowed.
+	ShouldLateInitialize() bool
+	// ShouldUpdate returns true if the Update action is allowed.
+	ShouldUpdate() bool
+	// ShouldDelete returns true if the Delete action is allowed.
+	ShouldDelete() bool
+}
+
+// ManagementPoliciesResolver is used to perform management policy checks
+// based on the management policy and if the management policy feature is enabled.
+type ManagementPoliciesResolver struct {
+	managementPolicyEnabled     bool
+	supportedManagementPolicies []sets.Set[xpv1.ManagementAction]
+	managementPolicy            sets.Set[xpv1.ManagementAction]
```

Comment:
```suggestion
	managementPolicies            sets.Set[xpv1.ManagementAction]
```

just to be consistent